lib/ansible/plugins/action/__init__.py

```diff
@@ -918,12 +918,16 @@ def _parse_returned_data(self, res):
                 if res['stderr'].startswith(u'Traceback'):
                     data['exception'] = res['stderr']
 
+            # The default
+            data['msg'] = "MODULE FAILURE"
+
             # try to figure out if we are missing interpreter
-            if self._used_interpreter is not None and '%s: No such file or directory' % self._used_interpreter.lstrip('!#') in data['module_stderr']:
-                data['msg'] = "The module failed to execute correctly, you probably need to set the interpreter."
-            else:
-                data['msg'] = "MODULE FAILURE"
+            if self._used_interpreter is not None:
+                match = '%s: No such file or directory' % self._used_interpreter.lstrip('!#')
+                if match in data['module_stderr'] or match in data['module_stdout']:
```

Comment:
```suggestion
                if match in (data['module_stderr'], data['module_stdout']):
```

Reply:
wont that match only if the string is exactly what stdout or stderr contain? we need to do partial match on the contents of stderr or stdout

Reply:
Yes, you are right. I didn't notice that we want to match partial string as well.